File: problems/same-tree/recursion.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        if (p is null && q is null) return true;
        if (p is null && q is not null) return false;
        if (p is not null && q is null) return false;

        if (p.val != q.val) return false;
        return IsSameTree(p.left, q.left) && IsSameTree(p.right, q.right);
    }
}

File: problems/path-sum/recursion.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if root is None:
            return False
        print(root.val)
        remainingSum = targetSum - root.val

        if root.left is None and root.right is None:
            return remainingSum == 0

        return self.hasPathSum(root.left, remainingSum) or self.hasPathSum(root.right, remainingSum)